injection/sharedmain/main.go

```diff
@@ -55,32 +56,33 @@ func MainWithContext(ctx context.Context, component string, ctors ...injection.C
 }
 
 func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, ctors ...injection.ControllerConstructor) {
+	log.Printf("Registering %d clients", len(injection.Default.GetClients()))
+	log.Printf("Registering %d informer factories", len(injection.Default.GetInformerFactories()))
+	log.Printf("Registering %d informers", len(injection.Default.GetInformers()))
+	log.Printf("Registering %d controllers", len(ctors))
+
+	// Adjust our client's rate limits based on the number of controller's we are running.
+	cfg.QPS = float32(len(ctors)) * rest.DefaultQPS
+	cfg.Burst = len(ctors) * rest.DefaultBurst
+
+	ctx, informers := injection.Default.SetupInformers(ctx, cfg)
+	kubeClient := kubeclient.Get(ctx)
+
 	// Set up our logger.
-	loggingConfigMap, err := configmap.Load("/etc/config-logging")
+	loggingConfigMap, err := kubeClient.CoreV1().ConfigMaps(system.Namespace()).Get(logging.ConfigMapName(), metav1.GetOptions{})
```

Comment:
Could we attempt to read from the filesystem first and then fall back to the API call?

Reply:
Yes we can! Done that and pushed an update.